### DISCUSSION TITLE
Media queries/responsiveness

### DISCUSSION THREAD

**Discussion by rtymchyk** [mantinedev/mantine#34] [Q&A] [answered]
Forgive me if I missed this in the docs, but what is the recommended way of handling responsive layouts? It appears that the components props do not allow specification dependant on screen size.

Everything looks amazing by the way 👌

**Comment by cstrat** [comment]
There is a hook which lets you subscribe to a media query:

https://mantine.dev/hooks/use-media-query/

**Reply by rtymchyk** [reply]
Yeah I saw this, but this will not work with server side rendering since it depends on a clientside browser API. Was hoping for something like this https://styled-system.com/responsive-styles/

**Comment by rtivital** [comment, marked as answer]
Hi Roman, you can use [react-jss media queries](https://cssinjs.org/jss-syntax/?v=v10.6.0#media-queries) or use any other css solution and add styles via className. Mantine components in most cases do not depend on viewport width so there are no premade responsive styles
